BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift

```diff
@@ -385,26 +388,65 @@ public final class ConvivaAnalytics: NSObject {
             value: Int64(player.currentTime(.relativeTime) * 1_000)
         )
     }
-}
 
-private extension Ad {
-    var adInfo: [String: Any] {
+    private func buildAdInfo(adStartedEvent: AdStartedEvent) -> [String: Any] {
         var adInfo = [String: Any]()
+
+        adInfo["c3.ad.id"] = notAvailable
+        adInfo["c3.ad.system"] = notAvailable
+        adInfo["c3.ad.mediaFileApiFramework"] = notAvailable
+        adInfo["c3.ad.firstAdSystem"] = notAvailable
+        adInfo["c3.ad.firstAdId"] = notAvailable
+        adInfo["c3.ad.firstCreativeId"] = notAvailable
+
         adInfo["c3.ad.technology"] = "Client Side"
+        if adStartedEvent.clientType == AdSourceType.ima {
```

Comment:
You don't need the enum type in Swift.

```suggestion
        if adStartedEvent.clientType == .ima {
```